### PR TITLE
[ROCm][GHA] split 4 GPU hosts across two runners

### PR DESCRIPTION
**Pull request by jeffdaily**
Examine the runner name.  If it ends with "-2" to indicate the second runner on the host, the docker run arguments will select the last two GPUs.  Otherwise, select the first two GPUs on the host.

### CODE REVIEW

File: .github/actions/setup-rocm/action.yml [hunk 1]
```diff
@@ -54,4 +54,8 @@ runs:
     - name: ROCm set GPU_FLAG
       shell: bash
       run: |
-        echo "GPU_FLAG=--device=/dev/mem --device=/dev/kfd --device=/dev/dri --group-add video --group-add daemon" >> "${GITHUB_ENV}"
+        if [[ ${{ runner.name }} == *-2 ]]; then
```
Comment (janeyx99) [inline .github/actions/setup-rocm/action.yml:57]: Can you add a comment here explaining what's happening?

**Comment by janeyx99** [comment]
This is quite an intriguing idea! Where will the logic regarding splitting each machine into 2 runners happen/where will the -2 appear?

**Comment by jeffdaily** [comment]
> This is quite an intriguing idea! Where will the logic regarding splitting each machine into 2 runners happen/where will the -2 appear?

The runners that are already registered will not change.  No new labels, etc.  We will register new runners on the hosts with 4 GPUs and name them with the "-2" suffix.  For example, runner `worker-rocm-amd-18` is registered on host `jenkins-worker-rocm-amd-18`.  That host would have another runner registered with the name `worker-rocm-amd-18-2`.  

There is nothing really separating the two runners.  By default, they both will have full access to host resources (cpu, memory, gpus).  This PR updates the docker run command in the workflow to use the runner name to determine which GPUs are selected.  Using the arg `--device=/dev/dri/renderD128` utilizes linux cgroups to separate the GPUs in the docker containers -- the two containers cannot use each others GPUs accidentally.

File: .github/actions/setup-rocm/action.yml (hunk 1 above)
Comment (janeyx99) [reply .github/actions/setup-rocm/action.yml:57]: I meant in the code haha, I understood the concept from your PR body, but a code reader may be confused.

**Review by janeyx99 — APPROVED** [review]
approving so you can land when you are ready

**Pull request closed without merging**